File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        
        for(int i=0; i<nums.length; i++){
            if(map.get(nums[i]) == null) {
                map.put(nums[i], 0);
            }
        }
        
        int answer = Math.min(nums.length/2, map.keySet().size());
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <algorithm>
#include <set>
using namespace std;

int solution(vector<int> nums)
{
  set<int> s;
  for(int num : nums) s.insert(num);
    int answer = min(s.size(), nums.size()/2);
    return answer;
}